ast/output_test.go

```diff
@@ -0,0 +1,215 @@
+package ast
+
+import (
+	"testing"
+)
+
+func TestOutput_type(t *testing.T) {
+	testCases := []struct{
+		Name string
+		Output *Output
+		Scope Scope
+		ReturnType Type
+		ShouldError bool
+	}{
```

Comment:
These tests are glorious! Nice add, and love that you included test names. Makes it wayyyy easier to understand the intent of the tests and therefore the intended behavior of the code.